ddocs/medic-db/medic-client/views/reports_by_freetext/map.js

```diff
@@ -15,32 +18,33 @@ function(doc) {
     if (!key || !value) {
       return;
     }
-    key = key.toLowerCase();
-    if (skip.indexOf(key) !== -1 || /_date$/.test(key)) {
-      return;
-    }
+
     if (typeof value === 'string') {
       value = value.toLowerCase();
       value.split(/\s+/).forEach(function(word) {
         emitMaybe(word, reportedDate);
       });
     }
-    if (typeof value === 'number' || typeof value === 'string') {
-      emitMaybe(key + ':' + value, reportedDate);
-    }
   };
 
-  if (doc.type === 'data_record' && doc.form) {
-    Object.keys(doc).forEach(function(key) {
-      emitField(key, doc[key], doc.reported_date);
-    });
+  var include = ['patient_id', 'patient_uuid', 'place_id', 'case_id', 'patient_name'];
+  include.forEach(function(key) {
+    emitField(key, doc[key], doc.reported_date);
+
     if (doc.fields) {
-      Object.keys(doc.fields).forEach(function(key) {
-        emitField(key, doc.fields[key], doc.reported_date);
-      });
-    }
-    if (doc.contact && doc.contact._id) {
-      emitMaybe('contact:' + doc.contact._id.toLowerCase(), doc.reported_date);
+      emitField(key, doc.fields[key], doc.reported_date);
     }
+  });
+
+  if (doc.contact && doc.contact._id) {
+    emitMaybe('contact:' + doc.contact._id.toLowerCase(), doc.reported_date);
+  }
+
+  if (doc.case_id) {
+    emitMaybe('case_id:' + doc.case_id, doc.reported_date);
```

Comment:
Looks like the `emitMaybe` function does not check the `value` parameter at all.  Do we care?  What happens if we `emit` an `undefined` value?  Maybe it would be better to still call the `emitField` function for all these special cases...

Reply:
We already had this bit before that emitted `doc.reported_date`
```ts
if (doc.contact && doc.contact._id) {
  emitMaybe('contact:' + doc.contact._id.toLowerCase(), doc.reported_date);
}
```
It worked this far so I think it's safe to assume `doc.reported_date` is always defined and we can use `emitMaybe()` for `case_id` as well

Reply:
:facepalm: Looking back and this now, I am not sure exactly what I was thinking when I left that comment...  You are correct that the logic seems good!